clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java

```diff
@@ -529,10 +529,9 @@ public ConsumerGroupHeartbeatRequestData buildRequestData() {
             // MemberEpoch - always sent
             data.setMemberEpoch(membershipManager.memberEpoch());
 
-            // InstanceId - only sent if has changed since the last heartbeat
-            // Always send when leaving the group as a static member
+            // InstanceId - send when leaving the group as a static member
             membershipManager.groupInstanceId().ifPresent(groupInstanceId -> {
-                if (!groupInstanceId.equals(sentFields.instanceId) || membershipManager.memberEpoch() == ConsumerGroupHeartbeatRequest.LEAVE_GROUP_STATIC_MEMBER_EPOCH) {
+                if (membershipManager.memberEpoch() == ConsumerGroupHeartbeatRequest.LEAVE_GROUP_STATIC_MEMBER_EPOCH) {
                     data.setInstanceId(groupInstanceId);
                     sentFields.instanceId = groupInstanceId;
```

Comment:
it seems `sentFields.instanceId` is useless after this PR. maybe we should remove it?

Reply:
I don't understand enough about this part of the code to judge...

Reply:
It is never read after this PR, so I feel it is safe to remove it. Otherwise, that will result in another warning about "never read".

Reply:
Agree we could simplify and remove the `instanceId` from the `sentFields` like @chia7712 suggested. I don't see the need to track it as it's not something that will change and we just simply need to send the value we have.